sonar-db/src/main/java/org/sonar/db/permission/PermissionQuery.java

```diff
@@ -19,106 +19,98 @@
+  private final String searchQueryToSql;
+  private final boolean withPermissionOnly;
+  private final List<String> logins;
+  private final List<String> groupNames;
 
-  // for internal use in MyBatis
-  final String searchSql;
-
-  // max results per page
   private final int pageSize;
-  // index of selected page. Start with 1.
-  private final int pageIndex;
-  // offset. Starts with 0.
   private final int pageOffset;
 
   private PermissionQuery(Builder builder) {
     this.permission = builder.permission;
-    this.component = builder.component;
+    this.withPermissionOnly = builder.withPermissionOnly;
+    this.componentUuid = builder.componentUuid;
     this.template = builder.template;
-    this.membership = builder.membership;
-    this.search = builder.search;
-    this.searchSql = searchToSql(search);
-
+    this.searchQuery = builder.searchQuery;
+    this.searchQueryToSql = builder.searchQuery == null ? null : buildLikeValue(builder.searchQuery, WildcardPosition.BEFORE_AND_AFTER).toLowerCase(Locale.ENGLISH);
     this.pageSize = builder.pageSize;
-    this.pageIndex = builder.pageIndex;
-    this.pageOffset = (builder.pageIndex - 1) * builder.pageSize;
+    this.pageOffset = offset(builder.pageIndex, builder.pageSize);
+    this.logins = builder.logins == null ? Collections.emptyList() : ImmutableList.copyOf(builder.logins);
+    this.groupNames = builder.groupNames == null ? Collections.emptyList() : ImmutableList.copyOf(builder.groupNames);
   }
 
-  private static String searchToSql(@Nullable String s) {
-    String sql = null;
-    if (s != null) {
-      sql = StringUtils.replace(StringUtils.upperCase(s), "%", "/%");
-      sql = StringUtils.replace(sql, "_", "/_");
-      sql = "%" + sql + "%";
-    }
-    return sql;
+  @CheckForNull
+  public String getPermission() {
+    return permission;
   }
 
-  public String permission() {
-    return permission;
+  public boolean withPermissionOnly() {
+    return withPermissionOnly;
   }
 
-  /**
-   * Used only for permission template
-   */
   public String template() {
     return template;
   }
 
-  /**
-   * Used on project permission
-   */
   @CheckForNull
-  public String component() {
-    return component;
+  public String getComponentUuid() {
+    return componentUuid;
   }
 
   @CheckForNull
-  public String membership() {
-    return membership;
+  public String getSearchQuery() {
+    return searchQuery;
   }
 
   @CheckForNull
-  public String search() {
-    return search;
+  public String getSearchQueryToSql() {
+    return searchQueryToSql;
   }
 
-  public int pageSize() {
+  public int getPageSize() {
     return pageSize;
   }
 
-  public int pageOffset() {
+  public int getPageOffset() {
     return pageOffset;
   }
 
-  public int pageIndex() {
-    return pageIndex;
+  public List<String> getLogins() {
+    return logins;
+  }
+
+  public List<String> getGroupNames() {
+    return groupNames;
   }
 
   public static Builder builder() {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUndocumentedApi)